nautobot/extras/models/computed_fields.py

```diff
@@ -0,0 +1,101 @@
+from collections import OrderedDict
+
+from django import forms
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.core.serializers.json import DjangoJSONEncoder
+from django.db import models
+
+# from nautobot.extras.models import ComputedField, CustomField
+# from nautobot.extras.models import ComputedField
+from django.urls import reverse
+from jinja2 import TemplateError
+
+from nautobot.core.models import BaseModel
+from nautobot.extras.models import ChangeLoggedModel
+from nautobot.extras.utils import extras_features, FeatureQuery, TemplateRenderer
+from nautobot.utilities.querysets import RestrictedQuerySet
+
+
+class ComputedFieldManager(models.Manager.from_queryset(RestrictedQuerySet)):
+    use_in_migrations = True
+
+    def get_for_model(self, model):
+        """
+        Return all CustomFields assigned to the given model.
+        """
+        content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
+        return self.get_queryset().filter(content_type=content_type)
+
+
+@extras_features("graphql")
+class ComputedField(BaseModel, ChangeLoggedModel):
+    """
+    A custom link to an external representation of a Nautobot object. The link text and URL fields accept Jinja2 template
+    code to be rendered with an object as context.
+    """
+
+    content_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        limit_choices_to=FeatureQuery("computed_fields"),
+    )
+    name = models.SlugField(max_length=100, unique=True, help_text="Internal field name")
```

Comment:
Might not be a bad idea to have a `slug` as well, and have things like `get_computed_fields` index the dict by `slug` rather than by `name`? Trying to head off future issues along the lines of https://github.com/nautobot/nautobot/issues/464.